### PR TITLE
For #942: Fixed `TextEnvelope.equals()` misbehavior.

### PR DESCRIPTION
**Pull request by paulodamaso**
For #942:
- Fixed `TextEnvelope.equals()` misbehavior;
- Fixed `TextEnvelopeTests` tests regarding `TextEnvelope.equals()`;
- Added puzzle for refactoring and making code more elegant.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#952](https://github.com/yegor256/cactoos/issues/952) is now in scope, role is `REV`

File: src/main/java/org/cactoos/text/TextEnvelope.java [hunk 1]
```diff
@@ -73,8 +73,23 @@ public final int hashCode() {
...
+    // this method so we can get rid of these smelly things.
+    //
     @Override
+    @SuppressWarnings("PMD.OnlyOneReturn")
     public final boolean equals(final Object obj) {
-        return new UncheckedScalar<>(this.origin).value().equals(obj);
+        if (this == obj) {
+            return true;
+        }
+        if (!(obj instanceof TextEnvelope)) {
```
Comment (llorllale) [inline src/main/java/org/cactoos/text/TextEnvelope.java]: @paulodamaso let's make program against the `Text` interface, not the implementation

**Comment by llorllale** [comment]
@paulodamaso left you a comment

Also, fix the puzzle's format.

**Comment by paulodamaso** [comment]
@llorllale Thanks for the review, corrections applied, please take a look

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#952` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#952](https://github.com/yegor256/cactoos/issues/952) is now out of scope

**Pull request merged**